docs/CLI.md

```diff
@@ -0,0 +1,41 @@
+# Dagger CLI
+
+> This is currently a throwaway markdown file for collaborating in a PR. Could
+> be turned into a doc but that's not the current priority.
+
+## Things you can do
+
+Unless otherwise specified, all commands would 
+
+* run tests/checks
+    * TODO decide between `test` or `check`
+        * `test` is arguably more familiar and guessable, but not every check
+          will be a test suite. It could also be a linter, or whatever the user
+          wants to run.
+        * Maybe we should just go with `test` anyway, favoring familiarity over
+          technical correctness? Arguably non-test-sutie checks are just
+          integration tests. :grin:
+    * `dagger check` - run all checks
+    * `dagger check engine` - run a named check
+    * `dagger check engine -- -run Services` - run a named check, passing args
+        * Needs a standard way to pass argv to check callbacks.
+    * `dagger test` - alternative
+    * `dagger test engine` - run a named check
+    * `dagger test engine -- -run Services` - run a named check, passing args
+        * Needs a standard way to pass argv to check callbacks.
+* build artifacts
+    * `dagger build`
+* list available artifacts
+    * `dagger artifacts`
+* extend your environment with another environment
+    * `dagger use go` - universe environment
+    * `dagger use ./go/` - local environment
+    * `dagger use github.com/vito/progrock@main` - git environment
+    * `dagger use github.com/vito/progrock/ci@main` - git environment, `./ci` subdirectory
```

Comment:
I'm being sneaky here with the subpath, since it looks cleaner than `?dir=foo`.

I figured we can match `go get` here, but it turns out that uses a Go-specific `<meta>` tag these days to figure out the URL. We could respect that, I guess, but it feels like a strange coupling to Go.

Safest heuristic would probably be to do what Go did before 1.4, where it assumes user/repo. (Assuming [ChatGPT isn't hallucinating](https://chat.openai.com/share/9e4168d7-d3f6-4314-8aab-8abd2f81bf25).) Seems like that would cover the 99% case, since even most non-GitHub repos follow that scheme. Would make sense to have some kind of fallback though, like maybe supporting an explicit `?dir=` anyway.

Reply:
That would break very easily in GitLab where you can have groups within groups until you reach your repo.